### PR TITLE
D-Bus improvements v3

### PR DESCRIPTION
**Pull request by vcgomes**
Changes from last version:
 - Fixed comments from review

### CODE REVIEW

**Comment by vcgomes** [comment]
Updated version of https://github.com/solettaproject/soletta/pull/1123

File: src/lib/common/sol-bus.c [hunk 1]
```diff
@@ -397,50 +531,59 @@ _getall_properties(sd_bus_message *reply, void *userdata,
...
-    r = sol_ptr_vector_append(&_ctx.property_tables, t);
-    SOL_INT_CHECK_GOTO(r, < 0, fail_append);
+    if (!client->properties_changed) {
+        r = snprintf(matchstr, sizeof(matchstr),
+            "type='signal',"
+            "sender='%s',"
+            "interface='org.freedesktop.DBus.Properties',"
+            "member='PropertiesChanged'",
+            client->service);
+        SOL_INT_CHECK_GOTO(r, < 0, fail);
```
Comment (lpereira) [inline src/lib/common/sol-bus.c]: `snprintf()` will return `< 0` on overflow only in older implementations; it's prudent to check for `>= sizeof(matchstr)` as well.

Comment (vcgomes) [inline src/lib/common/sol-bus.c]: As I explained in earlier versions of this series, I have a guarantee that `client->service` is smaller than 255 bytes, so the overflow case is not possible.

**Comment by ceolin** [comment]
the serie seems pretty good, +1

**Comment by vcgomes** [comment]
Merged. Thanks.

**Pull request closed without merging**